packages/fxa-payments-server/src/routes/Product/IapRoadblock/index.tsx

```diff
@@ -20,17 +20,19 @@ export type IapRoadblockProps = {
   profile: Profile;
   customer: Customer;
   selectedPlan: Plan;
-  subscription: IapSubscription;
+  subscription?: IapSubscription;
+  code: string;
 };
 
 const getIapSubscriptionAppStoreL10Id = (
   s: IapRoadblockProps['subscription']
 ) => {
-  switch (s._subscription_type) {
-    case MozillaSubscriptionTypes.IAP_GOOGLE:
-      return 'brand-name-google-play';
-    case MozillaSubscriptionTypes.IAP_APPLE:
-      return 'brand-name-apple-app-store';
+  if (s && s._subscription_type === MozillaSubscriptionTypes.IAP_GOOGLE) {
```

Comment:
nit. what was the reason from changing away from a switch statement here? (Apologies if this has already been asked an answered.)

Reply:
I had to change the logic slightly in this file to accept an undefined subscription, as this roadblock used to only address already subscribed subscriptions and not upgrades (different product, same product set). When `s` (subscription) is undefined, the switch logic won't work and then nothing can get returned. This seemed like the cleanest way to address it vs wrapping the switch statements in an if. Please let me know if there is a more efficient way though 🙂 .